File: Songhay.Dashboard/ClientApp/src/app/songhay/player/video/you-tube/models/you-tube-snippet.ts
import { YouTubeResourceId } from './you-tube-resource-id';
import { YouTubeThumbnails } from './you-tube-thumbnails';

/**
 * defines YouTube thumb snippet
 *
 * @export
 * @interface YouTubeSnippet
 */
export interface YouTubeSnippet {
    /**
     * channel ID
     *
     * @type {string}
     * @memberof YouTubeSnippet
     */
    channelId: string;

    /**
     * channel Title
     *
     * @type {string}
     * @memberof YouTubeSnippet
     */
    channelTitle: string;

    /**
     * publish date
     *
     * @type {Date}
     * @memberof YouTubeSnippet
     */
    publishedAt: Date;

    /**
     * resource ID
     *
     * @type {YouTubeResourceId}
     * @memberof YouTubeSnippet
     */
    resourceId: YouTubeResourceId;

    /**
     * thumbnail metadata
     *
     * @type {YouTubeThumbnails}
     * @memberof YouTubeSnippet
     */
    thumbnails: YouTubeThumbnails;
}

File: Songhay.Player.YouTube/Models/YouTubeSnippet.fs
namespace Songhay.Player.YouTube.Models

open System

/// <summary>
/// defines YouTube thumb snippet
/// </summary>
type YouTubeSnippet = {

    channelId: string

    channelTitle: string

    description: string

    /// <summary>
    /// localized version of description and title
    /// </summary>
    localized: {| description: string; title: string |}

    playlistId: string

    position: int option

    publishedAt: DateTime

    resourceId: YouTubeResourceId option

    tags: string[] option

    thumbnails: YouTubeThumbnails

    title: string
}
